sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -449,9 +460,10 @@ struct _pi_context : _pi_object {
 
   // Store USM allocator context(internal allocator structures)
   // for USM shared and device allocations. There is 1 allocator context
-  // per each pair of (context, device) per each memory type.
-  std::unordered_map<pi_device, USMAllocContext> SharedMemAllocContexts;
```

Comment:
The sub-sub-devices are represented by unique `pi_device`: https://github.com/intel/llvm/blob/0e22fdd28acb26253cbf5a1048daa76cf739c309/sycl/plugins/level_zero/pi_level_zero.cpp#L1607

Why is that not sufficient?